src/System Application/App/Email/src/Message/EmailMessageImpl.Codeunit.al

```diff
@@ -599,6 +599,43 @@ codeunit 8905 "Email Message Impl."
         exit(GlobalEmailMessage.Get(MessageId));
     end;
 
+    procedure DeleteOrphanedMessages(StartMessageId: Guid; MessagesToIterate: Integer) NextMessageId: Guid
+    var
+        EmailMessage: Record "Email Message";
+        EmptyGuid: Guid;
+        MessageNo: Integer;
+    begin
+        EmailMessage.SetLoadFields(Id);
+        EmailMessage.SetFilter(Id, '>=%1', StartMessageId);
+        if not EmailMessage.FindSet() then
```

Comment:
This would lock all rows after that Id. Customer can run into locking issues, if they have a ton which need to be deleted but are also trying to send emails where processes are running outside of business hours.

Reply:
It only locks if there's a locktable. Furthermore this is exactly why partner can choose how many messages to iterate over, then commit etc. This way we also avoid having IcMs for iterating too many or if we iterate too few and it takes too long. They are in control.

Reply:
https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/methods-auto/record/record-findset-boolean-method
Set this parameter to true if you want to modify any records in the set; otherwise, set the parameter to false. If you set this parameter to true, then the records will be read with Updlock. If you set this parameter to false, then you can still modify the records in the set, but these updates will not be performed optimally. The default value is false.

So no lock is set (since default value is false), with that being said, we can still change readisolation to only read committed, since I think by default it reads uncommitted. Will add that.